cmd/swapd/wait.go

```diff
@@ -1,22 +1,23 @@
 package main
 
 import (
+	"context"
 	"fmt"
 	"os"
 	"os/signal"
 	"syscall"
 )
 
-func (d *daemon) wait() {
+func signalHandler(ctx context.Context, cancel context.CancelFunc) {
 	sigc := make(chan os.Signal, 1)
 	signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM)
 	defer signal.Stop(sigc)
 
 	select {
 	case <-sigc:
 		fmt.Println("signal interrupt, shutting down...")
-		d.cancel()
-	case <-d.ctx.Done():
+		cancel()
+	case <-ctx.Done():
 		fmt.Println("protocol complete, shutting down...")
```

Comment:
Is it intentional that these are using `fmt.Println` instead of `log.SomeLevel`?

Reply:
no, log should also be fine